manifests/network/interface.pp

```diff
@@ -79,7 +80,9 @@
       }
 
       bsd_interface { $if_name:
-        ensure => $ensure,
+        ensure  => $ensure,
+        parents => $parents,
+        require => File["/etc/hostname.${if_name}"],
```

Comment:
Though, now that I'm seeing it, the autorequire in the provider might handle this already.